java-support/tck/src/main/java/io/cloudstate/javasupport/tck/model/action/ActionTckModelBehavior.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2019 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.cloudstate.javasupport.tck.model.action;
+
+import akka.NotUsed;
+import akka.actor.ActorSystem;
+import akka.stream.javadsl.Flow;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+import io.cloudstate.javasupport.Context;
+import io.cloudstate.javasupport.ServiceCall;
+import io.cloudstate.javasupport.action.*;
+import io.cloudstate.javasupport.action.Effect;
+import io.cloudstate.tck.model.Action.*;
+import java.util.concurrent.CompletionStage;
+
+@Action
+public class ActionTckModelBehavior {
```

Comment:
Is this Entity different to the eventsourced TCK entities as it is named `Behaviour`(I think this is an Akka/Actor thing? I don't know Akka) as the eventsourced Entity is not named with this pattern `EventSourcedTckModelEntity`.

Reply:
Actions are not considered to be Entities, in the definitions we've been using. Where an Entity is primarily defined by having a distinct identity, associated with its id or key. I think the usual pattern will be to name actions with `Action`, and I originally had `ActionTckModelAction` and `ActionTwoAction` here, and then tried naming it with `Behaviour` instead to avoid the awkwardness. Happy to find a naming convention and use it here in the TCK too.